examples/templates/configuration/raw-keyrings/raw-rsa-keyring-encrypt-with-public-only.md

```diff
@@ -0,0 +1,188 @@
+       "is not": "secret",
+       "but adds": "useful metadata",
+       "that can help you": "be confident that",
+       "the data you are handling": "is what you think it is",
+   }
+   ```
+
+1. Generate the wrapping key.
+
+   ```python
+   # Generate an RSA private key to use with your keyring.
+   # In practice, you should get this key from a secure key management system such as an HSM.
+   #
+   # The National Institute of Standards and Technology (NIST) recommends a minimum of 2048-bit keys for RSA.
+   # https://www.nist.gov/publications/transitioning-use-cryptographic-algorithms-and-key-lengths
+   #
+   # Why did we use this public exponent?
+   # https://crypto.stanford.edu/~dabo/pubs/papers/RSA-survey.pdf
+   private_key = rsa.generate_private_key(public_exponent=65537, key_size=4096, backend=default_backend())
+   ```
+
+1. Derive the public key from the wrapping private key.
+
+   ```python
+   # Collect the public key from the private key.
+   public_key = private_key.public_key()
+   ```
+
+1. Create keyrings.
+
+   ```python
+   # The keyring determines how your data keys are protected.
+   #
+   # Create the encrypt keyring that only has access to the public key.
+   public_key_keyring = RawRSAKeyring(
+       # The key namespace and key name are defined by you
+       # and are used by the raw RSA keyring
+       # to determine whether it should attempt to decrypt
+       # an encrypted data key.
+       #
+       # https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/choose-keyring.html#use-raw-rsa-keyring
+       key_namespace="some managed raw keys",
+       key_name="my RSA wrapping key",
+       public_wrapping_key=public_key,
+       # The wrapping algorithm tells the raw RSA keyring
+       # how to use your wrapping key to encrypt data keys.
+       #
+       # We recommend using RSA_OAEP_SHA256_MGF1.
+       # You should not use RSA_PKCS1 unless you require it for backwards compatibility.
+       wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+   )
+
+   # Create the decrypt keyring that has access to the private key.
+   private_key_keyring = RawRSAKeyring(
+       # The key namespace and key name MUST match the encrypt keyring.
+       key_namespace="some managed raw keys",
+       key_name="my RSA wrapping key",
+       private_wrapping_key=private_key,
+       # The wrapping algorithm MUST match the encrypt keyring.
+       wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+   )
+   ```
+
+1. Encrypt plaintext data using public wrapping key.
+
+   ```python
+   # Encrypt your plaintext data using the encrypt keyring.
+   ciphertext, _encrypt_header = aws_encryption_sdk.encrypt(
+       source=source_plaintext, encryption_context=encryption_context, keyring=public_key_keyring
+   )
+   ```
+
+1. Compare ciphertext to plaintext.
+
+   ```python
+   # Demonstrate that the ciphertext and plaintext are different.
+   assert ciphertext != source_plaintext
```

Comment:
Again, I don't think we are going to want to do this for every language,
and some will look different than Python :(